eosdatadistribution/ndn/base.py

```diff
@@ -115,6 +115,11 @@ def __init__(self, name, face=None):
         self._callbackCount = 0
         self._responseCount = 0
 
+        if auto: self.start()
```

Comment:
This is broken, and that's kind of what I meant. Since you chain up first thing in the constructor, `start()` will run before the rest of the constructor runs, leading to undefined member variable errors.